modules/calib3d/src/calibration.cpp

```diff
@@ -3141,11 +3141,18 @@ static void collectCalibrationData( InputArrayOfArrays objectPoints,
 
     for( i = 0; i < nimages; i++ )
     {
-        ni = objectPoints.getMat(i).checkVector(3, CV_32F);
-        if( ni <= 0 )
+        Mat objectPoint = objectPoints.getMat(i);
+        ni = objectPoint.checkVector(3, CV_32F);
+        if ( objectPoint.empty() )
+            CV_Error(CV_StsBadSize, "objectPoints should not contain empty vector of vectors of points");
+        else if( ni <= 0 )
```

Comment:
BTW,
`CV_Error()` doesn't return (throws exception), so "else if" is not necessary here.

Something like this:
```
Mat objectPoint = objectPoints.getMat(i);
if (objectPoint.empty())
    CV_Error(...)
ni = objectPoint.checkVector(3, CV_32F);
if (ni <= 0)
    CV_Error(...)
```

Reply:
@alalek I think the `else if` makes more sense.
E.g. if the `objectPoint` is indeed empty, 2 errors will be printed:
1. `OpenCV Error: The input/output structure size is incorrect (objectPoints should not contain empty vector of vectors of points) in collectCalibrationData`
2. `OpenCV Error: Unsupported format or combination of formats (objectPoints should contain vector of vectors of points of type Point3f) in collectCalibrationData`

In this case i think it's better that only error no. 1 be printed.
What is your opinion about this?

Reply:
> makes more sense

I understand your point from the code.
But it is correct in case if `CV_Error()` prints message, but `CV_Error()`

> throws exception

So, there is no way to see both messages and `else if` is not required here.

Reply:
@alalek You're right. I will fix it.